sdk/mobile-center/src/test/java/com/microsoft/azure/mobile/MobileCenterTest.java

```diff
@@ -796,6 +796,27 @@ public void uncaughtExceptionHandler() {
         ShutdownHelper.shutdown(10);
     }
 
+
+    @Test
+    public void uncaughtExceptionHandlerTimeout() throws Exception {
+
+        /* It behaves the same on timeout. */
+        Semaphore semaphore = mock(Semaphore.class);
+        whenNew(Semaphore.class).withAnyArguments().thenReturn(semaphore);
+        when(semaphore.tryAcquire(anyLong(), any(TimeUnit.class))).thenReturn(false);
+        uncaughtExceptionHandler();
+    }
+
+    @Test
+    public void uncaughtExceptionHandlerInterrupted() throws Exception {
```

Comment:
What do you want to test here?
At least you have to verify `MobileCenterLog.warn`.